service/counter/user.go

```diff
@@ -0,0 +1,227 @@
+
+	userStatus2Label := map[model.UserAccountStatus]string{
+		model.UserAccountStatusDeactivated: "user-deactivated",
+		model.UserAccountStatusActive:      "user-active",
+		model.UserAccountStatusSuspended:   "user-suspended",
+	}
+	botStatus2Label := map[model.BotState]string{
+		model.BotInactive: "bot-inactive",
+		model.BotActive:   "bot-active",
+		model.BotPaused:   "bot-paused",
+	}
+	go func() {
+		for e := range hub.Subscribe(1, event.UserCreated, event.UserUpdated, event.BotCreated, event.BotStateChanged, event.BotDeleted).Receiver {
+			switch e.Topic() {
+			case event.UserCreated:
+				counter.userCounter[model.UserAccountStatusActive].inc()
+				counter.userTotalCounter.inc()
+				usersCounter.WithLabelValues("user-active").Inc()
+				usersCounter.WithLabelValues("user-total").Inc()
+				userID := e.Fields["user_id"].(uuid.UUID)
+				counter.userStatus[userID] = model.UserAccountStatusActive
+
+			case event.UserUpdated:
+				userID := e.Fields["user_id"].(uuid.UUID)
+				var newStatusInInt int
+				db.Model(&model.User{}).Where("id = ?", userID).Select("status").First(&newStatusInInt)
+				newStatus := model.UserAccountStatus(newStatusInInt)
+				preStatus := counter.userStatus[userID]
+
+				counter.userCounter[preStatus].dec()
+				counter.userCounter[newStatus].inc()
+				usersCounter.WithLabelValues(userStatus2Label[preStatus]).Dec()
+				usersCounter.WithLabelValues(userStatus2Label[newStatus]).Inc()
+				counter.userStatus[userID] = newStatus
+
+			case event.BotCreated:
+				botState := e.Fields["bot"].(*model.Bot).State
+
+				counter.botCounter[botState].inc()
+				counter.userCounter[model.UserAccountStatusActive].dec()
+				counter.botTotalCounter.inc()
+				counter.userTotalCounter.dec()
+
+				usersCounter.WithLabelValues(botStatus2Label[botState]).Inc()
+				usersCounter.WithLabelValues("user-active").Dec()
+				usersCounter.WithLabelValues("bot-total").Inc()
+				usersCounter.WithLabelValues("user-total").Dec()
+
+				createdBotID := e.Fields["bot_id"].(uuid.UUID)
+				counter.botStatus[createdBotID] = botState
+
+			case event.BotStateChanged:
+				changedBotID := e.Fields["bot_id"].(uuid.UUID)
+				newStatus := e.Fields["state"].(model.BotState)
+				preStatus := counter.botStatus[changedBotID]
+
+				counter.botCounter[preStatus].dec()
+				counter.botCounter[newStatus].inc()
+				usersCounter.WithLabelValues(botStatus2Label[preStatus]).Dec()
+				usersCounter.WithLabelValues(botStatus2Label[newStatus]).Inc()
+				counter.botStatus[changedBotID] = newStatus
+
+			case event.BotDeleted:
+				deletedBotID := e.Fields["bot_id"].(uuid.UUID)
+				deletedBotState := counter.botStatus[deletedBotID]
+
+				counter.botCounter[deletedBotState].dec()
+				counter.botTotalCounter.dec()
+				usersCounter.WithLabelValues(botStatus2Label[deletedBotState]).Dec()
+				usersCounter.WithLabelValues("bot-total").Dec()
+				counter.botStatus[deletedBotID] = model.BotState(-1)
```

Comment:
`delete` メソッドを使ってmapからキーを削除してしまっても良いと思います.

```suggestion
				delete(counter.botStatus, deletedBotId)
```